EpiAware/docs/src/getting-started/explainers/censored-obs.jl

```diff
@@ -0,0 +1,321 @@
+	using StatsPlots #plotting
+	using Turing, TuringBenchmarking #PPL
+end
+
+# ╔═╡ 8de5c5e0-6e95-11ef-1693-bfd465c8d919
+md"
+# Fitting distributions using `censored_pmf` and Turing PPL
+
+## Introduction
+
+### What are we going to do in this Vignette
+
+In this vignette, we'll demonstrate how to use `EpiAwareUtils.censored_pmf` in conjunction with the Turing PPL for Bayesian inference of epidemiological delay distributions. We'll cover the following key points:
+
+1. Simulating censored delay distribution data
+2. Fitting a naive model using Turing
+3. Evaluating the naive model's performance
+4. Fitting an improved model using `censored_pmf` functionality
+5. Comparing the `censored_pmf` model's performance to the naive model
+
+### What might I need to know before starting
+
+This note builds on the concepts introduced in the R/stan package [`primarycensoreddist`](https://github.com/epinowcast/primarycensoreddist), especially the [Getting Started with primarycensoreddist](https://primarycensoreddist.epinowcast.org/articles/fitting-dists-with-stan.html) vignette and assumes familiarity with using Turing tools as covered in the [Turing documentation](https://turinglang.org/).
+
+This note is generated using the `EpiAware` package locally via `Pkg.develop`, in the `EpiAware/docs` environment. It is also possible to install `EpiAware` using 
+
+```julia
+Pkg.add(url=\"https://github.com/CDCgov/Rt-without-renewal\", subdir=\"EpiAware\")
+```
+
+"
+
+# ╔═╡ 30dd9af4-b64f-42b1-8439-a890752f68e3
+md"
+The other dependencies are as follows:
+"
+
+# ╔═╡ c5704f67-208d-4c2e-8513-c07c6b94ca99
+md"
+## Simulating censored and truncated delay distribution data
+
+We'll start by simulating some censored and truncated delay distribution data.
+"
+
+# ╔═╡ aed124c7-b4ba-4c97-a01f-ff553f376c86
+Random.seed!(123) # For reproducibility
+
+# ╔═╡ 105b9594-36ce-4ae8-87a8-5c81867b1ce3
+# Define the true distribution parameters
+n = 1000
+
+# ╔═╡ 8aa9f9c1-d3c4-49f3-be18-a400fc71e8f7
+meanlog = 1.5
+
+# ╔═╡ 84bb3999-9f2b-4eaa-9c2d-776a86677eaf
+sdlog = 0.75
+
+# ╔═╡ 2bf6677e-ebe9-4aa8-aa91-f631e99669bb
+true_dist = LogNormal(meanlog, sdlog)
+
+# ╔═╡ aea8b28e-fffe-4aa6-b51e-8199a7c7975c
+# Generate varying pwindow, swindow, and obs_time lengths
+pwindows = rand(1:1, n)
+
+# ╔═╡ d231bd0c-165f-4973-a46f-f66991813ea7
+swindows = rand(1:1, n)
+
+# ╔═╡ 7522f05b-1750-4983-8947-ef70f4298d06
+obs_times = fill(10.0,n)
+
+# ╔═╡ a4f5e9b6-ff3a-48fa-aa51-0abccb9c7bed
+#Sample secondary time relative to beginning of primary censor window respecting the right-truncation
+samples = map(pwindows, swindows, obs_times) do pw, sw, ot
+	P = rand() * pw # Primary event time 
+	T = rand(truncated(true_dist; upper= ot - P))
+end
+
+# ╔═╡ 0b5e96eb-9312-472e-8a88-d4509a4f25d0
+# Generate samples
+delay_counts = mapreduce(vcat, samples, pwindows, swindows, obs_times) do T, pw, sw, ot
+	DataFrame(
+		pwindow = pw, 
+		swindow = sw, 
+		obs_time = ot, 
+		observed_delay = T ÷ sw .|> Int,
```

Comment:
is this only correct for integer sw?

Reply:
No its correct for floats e.g. 

```julia
4.5 ÷ 0.7512
# 5.0
4.5 / 0.7512
# 5.9904153354632586
```

Reply:
is that correct though? We want to floor by window so that for i.e window 0.5 you get 0, 0.5, 1, 1.5 etc

Reply:
Right. Gotcha.

Reply:
We want

``julia
(T ÷ sw) * sw 
```

Reply:
Put that in now

Reply:
But we want 

```
Int(T ÷ sw) + 1
```

To link to element of log-pmf array (also done)